doc/user-guide/push-notifications/Push-notifications.md

```diff
@@ -0,0 +1,175 @@
+## How to set up Push Notifications
+
+Push notifications are the bread and butter of the modern mobile experience, and MongooseIM has
+support for them. When used together with [MongoosePush][], you get out-of-the-box push
+notifications for FCM (**F**irebase **C**loud **M**essaging) and APNS (**A**pple **P**ush
+**N**otification **S**ervice) providers. And it's easy to extend it to any other protocols of your
+choice.
+
+Server side push notification support is fully compliant with [XEP-0357 Push Notifications][XEP-0357],
+which defines several components that need to work together in order to provide clients with working
+push notifications. However, there's just one non-optimal detail required by the aforementioned XEP:
+that push notifications being a PubSub service — we can do better than that.
+
+All push notification mechanisms use [mod_event_pusher_push][] as a backend implementation,
+read [this][mod_event_pusher] to know more about it.
+
+While the whole setup can be incredibly extensible, we see three straightforward uses of it.
+
+### MongooseIM as a PubSub-less XMPP server
+
+ ![](PushNotifications-no_pubsub.png)
+
+PubSub is completely bypassed and the clients don't need to create a push node.
+They only have to provide the virtual PubSub address.
+
+A direct connection to a push service (e.g. MongoosePush) must be configured on the same MongooseIM
+node, checkout [this][WithMongoosePush] tutorial on how to setup [MongoosePush][].
+
+If compatibility with client libraries is not an issue, this is a really good approach.
+PubSub is completely bypassed and the clients don't need to create a push node — if they attempt to
+do so, the server would respond with an error stanza, hence the incompatibility with [XEP-0357][].
+They only have to provide the virtual PubSub address.
+
+```Erlang
+{mod_event_pusher, [
+    {backends, [
+        {push, [
+            {backend, mnesia}, % optional
+            {wpool, [{workers, 200}]}, % optional
+            {plugin_module, mod_event_pusher_push_plugin_defaults}, % optional, read more [here][mod_event_pusher_push_plugin]
```

Comment:
Are you sure the link will work in the code block?

Reply:
Damn, good catch, it won't! 😅